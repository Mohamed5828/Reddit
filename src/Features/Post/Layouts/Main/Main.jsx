// Import styled components
import { Title, Container } from "./Main.styled";

// Import components and layouts
import ChooseCommunity from "Features/Post/Components/ChooseCommunity/ChooseCommunity";
import Tabs from "Features/Post/Layouts/Tabs/Tabs";

// Import hooks
import { useState } from "react";

// Import api
import axios from "API/axios";
import useFetchFunction from "Hooks/useFetchFunction";
import { useCreatePostTitle } from "Features/Post/Contexts/createPostTitle";
import { useCreatePostText } from "Features/Post/Contexts/createPostText";
import { useCreatePostFlags } from "Features/Post/Contexts/createPostFlags";
import { useCreatePostFlairs } from "Features/Post/Contexts/createPostFlairs";
import { useCreatePostAttachments } from "Features/Post/Contexts/createPostAttachments";

/**
 * Main component (the main section in the create post page)
 *
 * @returns {React.Component} Main component (the main section in the create post page)
 */
const Main = () => {
  // State to store the selected community
  const [selectedDestination, setSelectedDestination] = useState(null);

  const [data, error, isLoading, dataFetch] = useFetchFunction();

  const { createPostTitle, setCreatePostTitle } = useCreatePostTitle();
  const { createPostText, setCreatePostText } = useCreatePostText();
  const { createPostFlags, setCreatePostFlags } = useCreatePostFlags();
  const { createPostFlairs, setCreatePostFlairs } = useCreatePostFlairs();
  const { createPostAttachments, setCreatePostAttachments } =
    useCreatePostAttachments();
  const submitPost = () => {
    // console.log({
    //   title: createPostTitle,
    //   originalText: createPostText,
    //   ...createPostFlags,
    //   ...createPostFlairs,
    //   attachments: createPostAttachments,
    // });
    dataFetch({
      axiosInstance: axios,
      method: "post",
      url: "/submit/",
      requestConfig: {
        data: {
          title: createPostTitle,
          originalText: createPostText,
          ...createPostFlags,
          ...createPostFlairs,
          attachments: createPostAttachments,
        },
      },
    });
  };
  return (
    <Container>
      <Title>
        <h4>Create a post</h4>
      </Title>
      <ChooseCommunity
        selectedDestination={selectedDestination}
        setSelectedDestination={setSelectedDestination}
      />
      <Tabs submitPost={submitPost} selectedDestination={selectedDestination} />
    </Container>
  );
};

export default Main;
