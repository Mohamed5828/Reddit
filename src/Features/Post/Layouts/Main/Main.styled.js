// Import styled
import styled from "styled-components";

// Main container for the main section
export const Container = styled.div`
  width: 800px;

  /* Disappear when width is less than 960px */
  @media (max-width: 960px) {
    width: 90%;
  }

  /* Disappear when width is less than 960px */
  @media (max-width: 640px) {
    width: 100%;
  }
`;

// Title container
export const Title = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.lineColor.primary};
  height: fit-content;
  margin-top: 25px;
  h4 {
    font-size: 18px;
  }
`;
