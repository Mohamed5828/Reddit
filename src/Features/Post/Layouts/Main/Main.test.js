import { shallow } from "enzyme";
import React from "react";

// Import components
import Main from "./Main";

describe("Main section", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Main />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should contains header", () => {
    expect(wrapper.find("h4").length).toBe(1);
    expect(wrapper.find("h4").text()).toBe("Create a post");
  });
});
