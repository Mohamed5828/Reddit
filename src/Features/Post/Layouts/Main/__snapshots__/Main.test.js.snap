// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main section should render without crashing 1`] = `
<styled.div>
  <styled.div>
    <h4>
      Create a post
    </h4>
  </styled.div>
  <ChooseCommunity
    selectedDestination={null}
    setSelectedDestination={[Function]}
  />
  <Tabs
    selectedDestination={null}
    submitPost={[Function]}
  />
</styled.div>
`;
